### PR TITLE
feat: Add HasStyle interface to Dialog #2856

### LINKED ISSUE #2856: Add HasStyle interface to Dialog
## Description

1. Add missing `HasStyle` interface to `Dialog`
2. Add unit tests to check the interface - [example](https://github.com/vaadin/flow-components/blob/master/vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/test/java/com/vaadin/flow/component/checkbox/tests/HasLabelTest.java)
3. Override methods to set classes on the [overlay element](https://github.com/vaadin/web-components/blob/b6c2cbd894db62116f9b9ccc9bcf0de3bb7186ac/packages/dialog/src/vaadin-dialog.js#L183)

Part of https://github.com/vaadin/platform/issues/2801

## See also

- [`HasStyle`](https://github.com/vaadin/flow/blob/master/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java) implementation in Flow
- [Unit tests for `HasStyle`](https://github.com/vaadin/flow/blob/master/flow-server/src/test/java/com/vaadin/flow/component/HasStyleTest.java)

### PR DESCRIPTION
**Pull request by krissvaa**
## Description

Partially implement 'HasStyles' interface for Dialog. 
Now class names can be added to dialog overlay.
Adding styles is not supported right now as logic will be a bit convoluted. 

Fixes #2856

## Type of change

- [ ] Bugfix
- [X] Feature

## Checklist

- [X] I have read the contribution guide: https://vaadin.com/docs-beta/latest/guide/contributing/overview/
- [X] I have added a description following the guideline.
- [X] The issue is created in the corresponding repository and I have referenced it.
- [X] I have added tests to ensure my change is effective and works as intended.
- [X] New and existing tests are passing locally with my change.
- [X] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 1]
```diff
@@ -708,4 +717,14 @@ private void attachComponentRenderer() {
         setDimension(ElementConstants.STYLE_MIN_HEIGHT, minHeight);
         setDimension(ElementConstants.STYLE_MAX_HEIGHT, maxHeight);
     }
+
+    /**
+     * @throws UnsupportedOperationException
+     *             Adding styles to dialog overlay are not supported
```
Comment (web-padawan) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:723]: Let's align with the description of the same method in `ConfirmDialog`:

```suggestion
     *             Dialog does not support adding styles to overlay
```

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 2]
```diff
@@ -708,4 +717,14 @@ private void attachComponentRenderer() {
...
+     */
+    @Override
+    public Style getStyle() {
+        throw new UnsupportedOperationException();
```
Comment (web-padawan) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:727]: Let's add the description here too:

```suggestion
        throw new UnsupportedOperationException(
                "Dialog does not support adding styles to overlay");
```

File: vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java [hunk 3]
```diff
@@ -37,16 +37,19 @@
 import com.vaadin.flow.component.dependency.JsModule;
 import com.vaadin.flow.dom.Element;
 import com.vaadin.flow.dom.ElementConstants;
+import com.vaadin.flow.dom.Style;
+import com.vaadin.flow.function.SerializableConsumer;
```
Comment (web-padawan) [inline vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:41]: Is this import used anywhere?

Comment (DiegoCardoso) [reply vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java:41]: `Style` is being used but `SerializableConsumer` is not.

**Review by web-padawan — APPROVED** [review]

**Pull request merged**